### PR TITLE
Add SetThreadIdealProcessorEx PAL API

### PR DESCRIPTION
**Pull request by helloguo**
This PR adds SetThreadIdealProcessorEx PAL API for Linux.

### CODE REVIEW

**Comment by helloguo** [comment]
cc @janvorli

**Comment by janvorli** [comment]
@helloguo I am very sorry, this went completely out of my radar. Reviewing now.

**Comment by janvorli** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test please

File: src/pal/src/numa/numa.cpp [hunk 1]
```diff
@@ -867,3 +867,114 @@ VirtualAllocExNuma(
...
+                if (CPU_ISSET(i, &prevCpuSet))
+                {
+                    prevCpu = i;
+                    break;
+                }
+            }
+        }
+
+        _ASSERTE(prevCpu < g_possibleCpuCount);
+        lpPreviousIdealProcessor->Group = g_cpuToAffinity[prevCpu].Group;
```
Comment (janvorli) [inline src/pal/src/numa/numa.cpp]: The lpPreviousIdealProcessor should be updated after we read the lpIdealProcessor values. As it is now, if the same pointer is passed in both these args, we would overwrite the lpIdealProcessor with the previous one and then apply the new one, which means nothing would change.

Comment (janvorli) [reply src/pal/src/numa/numa.cpp]: Also, it would be good to set the Reserved member to 0.

**Comment by helloguo** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

**Pull request merged**